{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cyvcf2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "raw_vcf = \"/home/utsu/Desktop/WES-Pipeline/tests/output/nf.vep.vcf\"\n",
    "vcf = cyvcf2.VCF(raw_vcf)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "for h in vcf.header_iter():\n",
    "    try:\n",
    "        if h['ID'] == 'CSQ':\n",
    "            csq_keys: list = h['Description'].split('Format: ')[1].split('|')\n",
    "        else:\n",
    "            pass\n",
    "    except KeyError:\n",
    "        pass\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Basic variant info\n",
    "CHROM POS REF ALT \n",
    "\n",
    "# Variant info\n",
    "Consequence GeneSymbol HGNC_ID RefSeq HGVSc HGVSp EXON INTRON ENSG ENST BIOTYPE  \n",
    "\n",
    "# Variant call quality\n",
    "\n",
    "# Minor allele frequency\n",
    "gnomADe_AF gnomADg_AF\n",
    "\n",
    "# Gene-based\n",
    "pHaplo pTriplo\n",
    "\n",
    "# Prediction Scores\n",
    "REVEL AlphaMissense SIFT PolyPhen PrimateAI Enformer_SAD Enformer_SAR\n",
    "\n",
    "# Additional annotation\n",
    "Existing_variation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'PCDH11Y', 'Gene': 'ENSG00000099715', 'Feature_type': 'Transcript', 'Feature': 'ENST00000362095', 'BIOTYPE': 'protein_coding', 'EXON': '2/3', 'INTRON': '', 'HGVSc': 'ENST00000362095.5:c.3036T>G', 'HGVSp': 'ENSP00000355419.5:p.(Asn1012Lys)', 'cDNA_position': '3770', 'CDS_position': '3036', 'Protein_position': '1012', 'Amino_acids': 'N/K', 'Codons': 'aaT/aaG', 'Existing_variation': 'rs2563389&COSV53074468', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '15813', 'CANONICAL': 'YES', 'RefSeq': 'NM_032972.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.4452', 'gnomADe_AFR_AF': '0.7619', 'gnomADe_AMR_AF': '0.461', 'gnomADe_ASJ_AF': '0.6441', 'gnomADe_EAS_AF': '0.9948', 'gnomADe_FIN_AF': '0.8677', 'gnomADe_NFE_AF': '0.2927', 'gnomADe_OTH_AF': '0.4183', 'gnomADe_SAS_AF': '0.3736', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '30262602', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'SLC9B1P1', 'Gene': 'ENSG00000183704', 'Feature_type': 'Transcript', 'Feature': 'ENST00000331172', 'BIOTYPE': 'protein_coding', 'EXON': '2/6', 'INTRON': '', 'HGVSc': 'ENST00000331172.6:c.205C>G', 'HGVSp': 'ENSP00000331938.6:p.(Leu69Val)', 'cDNA_position': '205', 'CDS_position': '205', 'Protein_position': '69', 'Amino_acids': 'L/V', 'Codons': 'Cta/Gta', 'Existing_variation': 'rs75895346&COSV58651319', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '37492', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.98)', 'PolyPhen': 'benign(0.015)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.000', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'SLC9B1P1', 'Gene': 'ENSG00000183704', 'Feature_type': 'Transcript', 'Feature': 'ENST00000331172', 'BIOTYPE': 'protein_coding', 'EXON': '1/6', 'INTRON': '', 'HGVSc': 'ENST00000331172.6:c.170A>C', 'HGVSp': 'ENSP00000331938.6:p.(Tyr57Ser)', 'cDNA_position': '170', 'CDS_position': '170', 'Protein_position': '57', 'Amino_acids': 'Y/S', 'Codons': 'tAc/tCc', 'Existing_variation': 'rs78395530&COSV58653377', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '37492', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.066', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'PTPRM', 'Gene': 'ENSG00000173482', 'Feature_type': 'Transcript', 'Feature': 'ENST00000580170', 'BIOTYPE': 'protein_coding', 'EXON': '19/33', 'INTRON': '', 'HGVSc': 'ENST00000580170.1:c.2618C>G', 'HGVSp': 'ENSP00000463325.1:p.(Thr873Ser)', 'cDNA_position': '3655', 'CDS_position': '2618', 'Protein_position': '873', 'Amino_acids': 'T/S', 'Codons': 'aCt/aGt', 'Existing_variation': 'rs186897065', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '9675', 'CANONICAL': 'YES', 'RefSeq': 'NM_001105244.1', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.31)', 'PolyPhen': 'benign(0.019)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0001352', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.001843', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0.0001844', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.61597430706', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.984089840947998', 'pTriplo': '0.646041765680919', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'SOGA2', 'Gene': 'ENSG00000168502', 'Feature_type': 'Transcript', 'Feature': 'ENST00000359865', 'BIOTYPE': 'protein_coding', 'EXON': '10/17', 'INTRON': '', 'HGVSc': 'ENST00000359865.3:c.2371C>T', 'HGVSp': 'ENSP00000352927.3:p.(Arg791Trp)', 'cDNA_position': '2513', 'CDS_position': '2371', 'Protein_position': '791', 'Amino_acids': 'R/W', 'Codons': 'Cgg/Tgg', 'Existing_variation': 'rs144721445&COSV60405458', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '29121', 'CANONICAL': 'YES', 'RefSeq': 'NM_015210.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.18)', 'PolyPhen': 'possibly_damaging(0.533)', 'HGVS_OFFSET': '', 'gnomADe_AF': '6.431e-05', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0.000292', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.0003711', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.082', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.299095153809', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.807420282466303', 'pTriplo': '0.272951153719262', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'CEP192', 'Gene': 'ENSG00000101639', 'Feature_type': 'Transcript', 'Feature': 'ENST00000506447', 'BIOTYPE': 'protein_coding', 'EXON': '18/45', 'INTRON': '', 'HGVSc': 'ENST00000506447.1:c.3109G>C', 'HGVSp': 'ENSP00000427550.1:p.(Val1037Leu)', 'cDNA_position': '3189', 'CDS_position': '3109', 'Protein_position': '1037', 'Amino_acids': 'V/L', 'Codons': 'Gtg/Ctg', 'Existing_variation': 'rs773954388&COSV58061205&COSV58064498', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '25515', 'CANONICAL': 'YES', 'RefSeq': 'NM_032142.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.33)', 'PolyPhen': 'benign(0.003)', 'HGVS_OFFSET': '', 'gnomADe_AF': '4.776e-05', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.0006525', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1&1', 'PHENO': '0&1&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.131', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0832', 'am_protein_variant': 'V1037L', 'am_transcript_id': 'ENST00000506447.1', 'am_uniprot_id': 'Q8TEP8', 'PrimateAI': '0.264092862606', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.612711506089752', 'pTriplo': '0.408519970397376', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'ESCO1', 'Gene': 'ENSG00000141446', 'Feature_type': 'Transcript', 'Feature': 'ENST00000269214', 'BIOTYPE': 'protein_coding', 'EXON': '4/12', 'INTRON': '', 'HGVSc': 'ENST00000269214.5:c.53A>G', 'HGVSp': 'ENSP00000269214.4:p.(Lys18Arg)', 'cDNA_position': '991', 'CDS_position': '53', 'Protein_position': '18', 'Amino_acids': 'K/R', 'Codons': 'aAa/aGa', 'Existing_variation': 'rs185403786', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '24645', 'CANONICAL': 'YES', 'RefSeq': 'NM_052911.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'deleterious(0.03)', 'PolyPhen': 'benign(0.116)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0005725', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.007502', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0.0003381', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.068', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0894', 'am_protein_variant': 'K18R', 'am_transcript_id': 'ENST00000269214.5', 'am_uniprot_id': 'Q5FWF5', 'PrimateAI': '0.307756870985', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.886389212255227', 'pTriplo': '0.778464255662281', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'TCEB3B', 'Gene': 'ENSG00000206181', 'Feature_type': 'Transcript', 'Feature': 'ENST00000332567', 'BIOTYPE': 'protein_coding', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000332567.4:c.512C>G', 'HGVSp': 'ENSP00000331302.4:p.(Ser171Cys)', 'cDNA_position': '865', 'CDS_position': '512', 'Protein_position': '171', 'Amino_acids': 'S/C', 'Codons': 'tCt/tGt', 'Existing_variation': 'rs201818139', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '30771', 'CANONICAL': 'YES', 'RefSeq': 'NM_016427.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'deleterious(0)', 'PolyPhen': 'benign(0.174)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0002921', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.003984', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.058', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1123', 'am_protein_variant': 'S171C', 'am_transcript_id': 'ENST00000332567.4', 'am_uniprot_id': 'Q8IYF1', 'PrimateAI': '0.497368365526', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0656603300623359', 'pTriplo': '0.113112800413346', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'ZNF516', 'Gene': 'ENSG00000101493', 'Feature_type': 'Transcript', 'Feature': 'ENST00000443185', 'BIOTYPE': 'protein_coding', 'EXON': '4/8', 'INTRON': '', 'HGVSc': 'ENST00000443185.2:c.2368G>A', 'HGVSp': 'ENSP00000394757.2:p.(Val790Met)', 'cDNA_position': '2686', 'CDS_position': '2368', 'Protein_position': '790', 'Amino_acids': 'V/M', 'Codons': 'Gtg/Atg', 'Existing_variation': 'rs184632128', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '28990', 'CANONICAL': 'YES', 'RefSeq': 'NM_014643.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0.009)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.000522', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '2.9e-05', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.006454', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '3.543e-05', 'gnomADe_OTH_AF': '0.0004959', 'gnomADe_SAS_AF': '0.0001961', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.109', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0669', 'am_protein_variant': 'V790M', 'am_transcript_id': 'ENST00000443185.2', 'am_uniprot_id': 'Q92618', 'PrimateAI': '0.294588834047', 'Enformer_SAD': '-6.366e-05', 'Enformer_SAR': '-2.944e-05', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.997827363773828', 'pTriplo': '0.327513314031007', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'PQLC1', 'Gene': 'ENSG00000122490', 'Feature_type': 'Transcript', 'Feature': 'ENST00000397778', 'BIOTYPE': 'protein_coding', 'EXON': '5/6', 'INTRON': '', 'HGVSc': 'ENST00000397778.2:c.470C>T', 'HGVSp': 'ENSP00000380880.2:p.(Ala157Val)', 'cDNA_position': '653', 'CDS_position': '470', 'Protein_position': '157', 'Amino_acids': 'A/V', 'Codons': 'gCg/gTg', 'Existing_variation': 'rs150340109&COSV60844281', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '26188', 'CANONICAL': 'YES', 'RefSeq': 'NM_025078.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.12)', 'PolyPhen': 'benign(0.001)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1.997e-05', 'gnomADe_AFR_AF': '0.0001858', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '1.772e-05', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.078', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1156', 'am_protein_variant': 'A157V', 'am_transcript_id': 'ENST00000397778.2', 'am_uniprot_id': 'Q8N2U9', 'PrimateAI': '0.522643089294', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.887690359725472', 'pTriplo': '0.139067567134868', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'LIPI', 'Gene': 'ENSG00000188992', 'Feature_type': 'Transcript', 'Feature': 'ENST00000344577', 'BIOTYPE': 'protein_coding', 'EXON': '9/10', 'INTRON': '', 'HGVSc': 'ENST00000344577.2:c.1326A>T', 'HGVSp': 'ENSP00000343331.2:p.(Arg442Ser)', 'cDNA_position': '1352', 'CDS_position': '1326', 'Protein_position': '442', 'Amino_acids': 'R/S', 'Codons': 'agA/agT', 'Existing_variation': 'rs7283442&COSV60707668', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '18821', 'CANONICAL': 'YES', 'RefSeq': 'NM_198996.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '1', 'SIFT': 'tolerated(0.2)', 'PolyPhen': 'benign(0.012)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': 'benign', 'SOMATIC': '0&1', 'PHENO': '1&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.133', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.365808665752', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0563374085609673', 'pTriplo': '0.148938495220196', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AF165138.7', 'Gene': 'ENSG00000243440', 'Feature_type': 'Transcript', 'Feature': 'ENST00000400562', 'BIOTYPE': 'protein_coding', 'EXON': '6/7', 'INTRON': '', 'HGVSc': 'ENST00000400562.1:c.334A>G', 'HGVSp': 'ENSP00000383407.1:p.(Asn112Asp)', 'cDNA_position': '384', 'CDS_position': '334', 'Protein_position': '112', 'Amino_acids': 'N/D', 'Codons': 'Aat/Gat', 'Existing_variation': 'rs456067', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'Clone_based_vega_gene', 'HGNC_ID': '', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '0.9999', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.041', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0456', 'am_protein_variant': 'N112D', 'am_transcript_id': 'ENST00000400562.1', 'am_uniprot_id': 'A8MT66', 'PrimateAI': '0.350319206715', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.122478224237736', 'pTriplo': '0.197597726796547', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'N6AMT1', 'Gene': 'ENSG00000156239', 'Feature_type': 'Transcript', 'Feature': 'ENST00000303775', 'BIOTYPE': 'protein_coding', 'EXON': '5/6', 'INTRON': '', 'HGVSc': 'ENST00000303775.5:c.497A>G', 'HGVSp': 'ENSP00000303584.5:p.(Lys166Arg)', 'cDNA_position': '523', 'CDS_position': '497', 'Protein_position': '166', 'Amino_acids': 'K/R', 'Codons': 'aAa/aGa', 'Existing_variation': 'rs2205446&COSV58133557', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '16021', 'CANONICAL': 'YES', 'RefSeq': 'NM_013240.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0.001)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.020', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.3648635149', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.388477775805443', 'pTriplo': '0.173801265858151', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'N6AMT1', 'Gene': 'ENSG00000156239', 'Feature_type': 'Transcript', 'Feature': 'ENST00000303775', 'BIOTYPE': 'protein_coding', 'EXON': '5/6', 'INTRON': '', 'HGVSc': 'ENST00000303775.5:c.437A>G', 'HGVSp': 'ENSP00000303584.5:p.(Lys146Arg)', 'cDNA_position': '463', 'CDS_position': '437', 'Protein_position': '146', 'Amino_acids': 'K/R', 'Codons': 'aAa/aGa', 'Existing_variation': 'rs2205447&COSV58133566', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '16021', 'CANONICAL': 'YES', 'RefSeq': 'NM_013240.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.68)', 'PolyPhen': 'benign(0.005)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '0.9999', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.074', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.409727156162', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.388477775805443', 'pTriplo': '0.173801265858151', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'N6AMT1', 'Gene': 'ENSG00000156239', 'Feature_type': 'Transcript', 'Feature': 'ENST00000303775', 'BIOTYPE': 'protein_coding', 'EXON': '1/6', 'INTRON': '', 'HGVSc': 'ENST00000303775.5:c.100A>G', 'HGVSp': 'ENSP00000303584.5:p.(Asn34Asp)', 'cDNA_position': '126', 'CDS_position': '100', 'Protein_position': '34', 'Amino_acids': 'N/D', 'Codons': 'Aac/Gac', 'Existing_variation': 'rs1997607&COSV58133573', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '16021', 'CANONICAL': 'YES', 'RefSeq': 'NM_013240.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.115', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.77962833643', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.388477775805443', 'pTriplo': '0.173801265858151', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'C21orf62', 'Gene': 'ENSG00000205929', 'Feature_type': 'Transcript', 'Feature': 'ENST00000536776', 'BIOTYPE': 'protein_coding', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000536776.1:c.543T>A', 'HGVSp': 'ENSP00000444950.1:p.(Phe181Leu)', 'cDNA_position': '684', 'CDS_position': '543', 'Protein_position': '181', 'Amino_acids': 'F/L', 'Codons': 'ttT/ttA', 'Existing_variation': 'rs74617086&COSV66671590', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '1305', 'CANONICAL': 'YES', 'RefSeq': 'NM_019596.5&NM_001162496.2&NM_001162495.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.050', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.290392031655742', 'pTriplo': '0.324586794623979', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'ITSN1', 'Gene': 'ENSG00000205726', 'Feature_type': 'Transcript', 'Feature': 'ENST00000381318', 'BIOTYPE': 'protein_coding', 'EXON': '13/40', 'INTRON': '', 'HGVSc': 'ENST00000381318.3:c.1408G>T', 'HGVSp': 'ENSP00000370719.3:p.(Val470Leu)', 'cDNA_position': '1696', 'CDS_position': '1408', 'Protein_position': '470', 'Amino_acids': 'V/L', 'Codons': 'Gta/Tta', 'Existing_variation': 'rs566065722', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '6183', 'CANONICAL': 'YES', 'RefSeq': 'NM_003024.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '1', 'SIFT': 'tolerated(0.34)', 'PolyPhen': 'benign(0.009)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.051', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1134', 'am_protein_variant': 'V470L', 'am_transcript_id': 'ENST00000381318.3', 'am_uniprot_id': 'Q15811', 'PrimateAI': '0.740536689758', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.996413119403726', 'pTriplo': '0.996910442078786', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'SLC5A3', 'Gene': 'ENSG00000272962', 'Feature_type': 'Transcript', 'Feature': 'ENST00000608209', 'BIOTYPE': 'protein_coding', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000608209.1:c.148A>G', 'HGVSp': 'ENSP00000477068.1:p.(Thr50Ala)', 'cDNA_position': '660', 'CDS_position': '148', 'Protein_position': '50', 'Amino_acids': 'T/A', 'Codons': 'Aca/Gca', 'Existing_variation': 'rs8129891&COSV62967544', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'Uniprot_gn', 'HGNC_ID': '', 'CANONICAL': 'YES', 'RefSeq': 'NM_006933.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.15)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.713404893875', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.939070463690547', 'pTriplo': '0.854671892422637', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'SLC5A3', 'Gene': 'ENSG00000272962', 'Feature_type': 'Transcript', 'Feature': 'ENST00000608209', 'BIOTYPE': 'protein_coding', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000608209.1:c.1696C>A', 'HGVSp': 'ENSP00000477068.1:p.(Gln566Lys)', 'cDNA_position': '2208', 'CDS_position': '1696', 'Protein_position': '566', 'Amino_acids': 'Q/K', 'Codons': 'Caa/Aaa', 'Existing_variation': 'rs4817617&COSV62967548', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'Uniprot_gn', 'HGNC_ID': '', 'CANONICAL': 'YES', 'RefSeq': 'NM_006933.4', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.97)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.383928269148', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.939070463690547', 'pTriplo': '0.854671892422637', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'PIGP', 'Gene': 'ENSG00000185808', 'Feature_type': 'Transcript', 'Feature': 'ENST00000464265', 'BIOTYPE': 'protein_coding', 'EXON': '1/4', 'INTRON': '', 'HGVSc': 'ENST00000464265.1:c.25G>A', 'HGVSp': 'ENSP00000420037.1:p.(Ala9Thr)', 'cDNA_position': '249', 'CDS_position': '25', 'Protein_position': '9', 'Amino_acids': 'A/T', 'Codons': 'Gcg/Acg', 'Existing_variation': 'rs2507733&COSV61639845', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '3046', 'CANONICAL': 'YES', 'RefSeq': 'NM_153681.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '1', 'SIFT': 'tolerated_low_confidence(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '1', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '1', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': 'benign', 'SOMATIC': '0&1', 'PHENO': '1&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.028', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1268', 'am_protein_variant': 'A9T', 'am_transcript_id': 'ENST00000464265.1', 'am_uniprot_id': 'P57054', 'PrimateAI': '0.306646168232', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.52039589747455', 'pTriplo': '0.163768418654643', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'DSCAM', 'Gene': 'ENSG00000171587', 'Feature_type': 'Transcript', 'Feature': 'ENST00000400454', 'BIOTYPE': 'protein_coding', 'EXON': '6/33', 'INTRON': '', 'HGVSc': 'ENST00000400454.1:c.1079T>A', 'HGVSp': 'ENSP00000383303.1:p.(Val360Glu)', 'cDNA_position': '1557', 'CDS_position': '1079', 'Protein_position': '360', 'Amino_acids': 'V/E', 'Codons': 'gTg/gAg', 'Existing_variation': 'rs1190464657', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '3039', 'CANONICAL': 'YES', 'RefSeq': 'NM_001271534.1&NM_001389.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.08)', 'PolyPhen': 'probably_damaging(0.93)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.606', 'am_class': 'ambiguous', 'am_genome': 'hg19', 'am_pathogenicity': '0.5319', 'am_protein_variant': 'V360E', 'am_transcript_id': 'ENST00000400454.1', 'am_uniprot_id': 'O60469', 'PrimateAI': '0.759580552578', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.976693579322956', 'pTriplo': '0.696558238138578', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'KRTAP10-1', 'Gene': 'ENSG00000215455', 'Feature_type': 'Transcript', 'Feature': 'ENST00000400375', 'BIOTYPE': 'protein_coding', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000400375.1:c.477T>A', 'HGVSp': 'ENSP00000383226.1:p.(Asp159Glu)', 'cDNA_position': '522', 'CDS_position': '477', 'Protein_position': '159', 'Amino_acids': 'D/E', 'Codons': 'gaT/gaA', 'Existing_variation': 'rs62218857&COSV59951573', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '22966', 'CANONICAL': 'YES', 'RefSeq': 'NM_198691.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.43)', 'PolyPhen': 'benign(0.014)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.028', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1016', 'am_protein_variant': 'D159E', 'am_transcript_id': 'ENST00000400375.1', 'am_uniprot_id': 'P60331', 'PrimateAI': '0.239087313414', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0827269848530047', 'pTriplo': '0.185677783590483', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'KRTAP10-4', 'Gene': 'ENSG00000215454', 'Feature_type': 'Transcript', 'Feature': 'ENST00000400374', 'BIOTYPE': 'protein_coding', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000400374.3:c.475A>G', 'HGVSp': 'ENSP00000383225.3:p.(Ile159Val)', 'cDNA_position': '505', 'CDS_position': '475', 'Protein_position': '159', 'Amino_acids': 'I/V', 'Codons': 'Atc/Gtc', 'Existing_variation': 'rs79048509&COSV59952364', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20521', 'CANONICAL': 'YES', 'RefSeq': 'NM_198687.1', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.1973', 'gnomADe_AFR_AF': '0.2943', 'gnomADe_AMR_AF': '0.3514', 'gnomADe_ASJ_AF': '0.1133', 'gnomADe_EAS_AF': '0.3335', 'gnomADe_FIN_AF': '0.1384', 'gnomADe_NFE_AF': '0.1247', 'gnomADe_OTH_AF': '0.1874', 'gnomADe_SAS_AF': '0.2446', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.002', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.061', 'am_protein_variant': 'I159V', 'am_transcript_id': 'ENST00000400374.3', 'am_uniprot_id': 'P60372', 'PrimateAI': '0.237539067864', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.149599494425567', 'pTriplo': '0.155677609846175', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'ITGB2', 'Gene': 'ENSG00000160255', 'Feature_type': 'Transcript', 'Feature': 'ENST00000397850', 'BIOTYPE': 'protein_coding', 'EXON': '10/17', 'INTRON': '', 'HGVSc': 'ENST00000397850.2:c.1062A>T', 'HGVSp': 'ENSP00000380948.2:p.(Gln354His)', 'cDNA_position': '1515', 'CDS_position': '1062', 'Protein_position': '354', 'Amino_acids': 'Q/H', 'Codons': 'caA/caT', 'Existing_variation': 'rs235330&COSV56612183', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '6155', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '1', 'SIFT': 'deleterious(0.03)', 'PolyPhen': 'possibly_damaging(0.681)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.9999', 'gnomADe_AFR_AF': '1', 'gnomADe_AMR_AF': '1', 'gnomADe_ASJ_AF': '1', 'gnomADe_EAS_AF': '1', 'gnomADe_FIN_AF': '1', 'gnomADe_NFE_AF': '0.9998', 'gnomADe_OTH_AF': '1', 'gnomADe_SAS_AF': '1', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': 'benign', 'SOMATIC': '0&1', 'PHENO': '1&1', 'PUBMED': '34944636', 'NMD': '', 'REVEL': '0.435', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.471266269684', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.774532627192783', 'pTriplo': '0.50995293653065', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'OR11H12', 'Gene': 'ENSG00000257115', 'Feature_type': 'Transcript', 'Feature': 'ENST00000550708', 'BIOTYPE': 'protein_coding', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000550708.1:c.437T>A', 'HGVSp': 'ENSP00000449002.1:p.(Leu146His)', 'cDNA_position': '509', 'CDS_position': '437', 'Protein_position': '146', 'Amino_acids': 'L/H', 'Codons': 'cTc/cAc', 'Existing_variation': 'rs138051664&COSV73569057', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '30738', 'CANONICAL': 'YES', 'RefSeq': 'NM_001013354.1&NM_001197287.1', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.3377', 'gnomADe_AFR_AF': '0.4422', 'gnomADe_AMR_AF': '0.3112', 'gnomADe_ASJ_AF': '0.2117', 'gnomADe_EAS_AF': '0.4064', 'gnomADe_FIN_AF': '0.3927', 'gnomADe_NFE_AF': '0.3375', 'gnomADe_OTH_AF': '0.3479', 'gnomADe_SAS_AF': '0.2527', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.027', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0554', 'am_protein_variant': 'L146H', 'am_transcript_id': 'ENST00000550708.1', 'am_uniprot_id': 'B2RN74', 'PrimateAI': '0.258437037468', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0247380774831715', 'pTriplo': '0.0667648011933666', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'OR11H12', 'Gene': 'ENSG00000257115', 'Feature_type': 'Transcript', 'Feature': 'ENST00000550708', 'BIOTYPE': 'protein_coding', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000550708.1:c.802A>G', 'HGVSp': 'ENSP00000449002.1:p.(Ser268Gly)', 'cDNA_position': '874', 'CDS_position': '802', 'Protein_position': '268', 'Amino_acids': 'S/G', 'Codons': 'Agc/Ggc', 'Existing_variation': 'rs201551285&COSV73569053', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '30738', 'CANONICAL': 'YES', 'RefSeq': 'NM_001013354.1&NM_001197287.1', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.1507', 'gnomADe_AFR_AF': '0.07516', 'gnomADe_AMR_AF': '0.1522', 'gnomADe_ASJ_AF': '0.215', 'gnomADe_EAS_AF': '0.1754', 'gnomADe_FIN_AF': '0.09479', 'gnomADe_NFE_AF': '0.1659', 'gnomADe_OTH_AF': '0.1752', 'gnomADe_SAS_AF': '0.196', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.075', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0463', 'am_protein_variant': 'S268G', 'am_transcript_id': 'ENST00000550708.1', 'am_uniprot_id': 'B2RN74', 'PrimateAI': '0.248174086213', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0247380774831715', 'pTriplo': '0.0667648011933666', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'POTEG', 'Gene': 'ENSG00000222036', 'Feature_type': 'Transcript', 'Feature': 'ENST00000409832', 'BIOTYPE': 'protein_coding', 'EXON': '1/11', 'INTRON': '', 'HGVSc': 'ENST00000409832.3:c.226A>G', 'HGVSp': 'ENSP00000386971.3:p.(Ser76Gly)', 'cDNA_position': '278', 'CDS_position': '226', 'Protein_position': '76', 'Amino_acids': 'S/G', 'Codons': 'Agc/Ggc', 'Existing_variation': 'rs57000026&COSV69152662', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '33896', 'CANONICAL': 'YES', 'RefSeq': 'NM_001005356.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated_low_confidence(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.002962', 'gnomADe_AFR_AF': '0.005638', 'gnomADe_AMR_AF': '0.003131', 'gnomADe_ASJ_AF': '0.00205', 'gnomADe_EAS_AF': '0.00139', 'gnomADe_FIN_AF': '0.004705', 'gnomADe_NFE_AF': '0.002253', 'gnomADe_OTH_AF': '0.00456', 'gnomADe_SAS_AF': '0.004002', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.011', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0497', 'am_protein_variant': 'S76G', 'am_transcript_id': 'ENST00000409832.3', 'am_uniprot_id': 'Q6S5H5', 'PrimateAI': '0.454752922058', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.0213334093144957', 'pTriplo': '0.272641616262717', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'EXOC5', 'Gene': 'ENSG00000070367', 'Feature_type': 'Transcript', 'Feature': 'ENST00000413566', 'BIOTYPE': 'protein_coding', 'EXON': '1/18', 'INTRON': '', 'HGVSc': 'ENST00000413566.2:c.14C>T', 'HGVSp': 'ENSP00000389934.2:p.(Ala5Val)', 'cDNA_position': '374', 'CDS_position': '14', 'Protein_position': '5', 'Amino_acids': 'A/V', 'Codons': 'gCc/gTc', 'Existing_variation': 'rs1380340562&COSV55104352', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '10696', 'CANONICAL': 'YES', 'RefSeq': 'NM_006544.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.16)', 'PolyPhen': 'benign(0.011)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.141', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1816', 'am_protein_variant': 'A5V', 'am_transcript_id': 'ENST00000413566.2', 'am_uniprot_id': 'O00471', 'PrimateAI': '0.800988733768', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.973028664769889', 'pTriplo': '0.795217991680687', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'GALNT16', 'Gene': 'ENSG00000100626', 'Feature_type': 'Transcript', 'Feature': 'ENST00000337827', 'BIOTYPE': 'protein_coding', 'EXON': '3/16', 'INTRON': '', 'HGVSc': 'ENST00000337827.4:c.421C>T', 'HGVSp': 'ENSP00000336729.4:p.(Arg141Cys)', 'cDNA_position': '748', 'CDS_position': '421', 'Protein_position': '141', 'Amino_acids': 'R/C', 'Codons': 'Cgc/Tgc', 'Existing_variation': 'rs750373256', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '23233', 'CANONICAL': 'YES', 'RefSeq': 'NM_020692.2&NM_001168368.1', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'deleterious(0)', 'PolyPhen': 'probably_damaging(0.999)', 'HGVS_OFFSET': '', 'gnomADe_AF': '3.635e-05', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '2.948e-05', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '5.499e-05', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '4.47e-05', 'gnomADe_OTH_AF': '0.0001655', 'gnomADe_SAS_AF': '3.325e-05', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.664', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.841059565544', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.695177030783854', 'pTriplo': '0.584789575450005', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'ACOT2', 'Gene': 'ENSG00000119673', 'Feature_type': 'Transcript', 'Feature': 'ENST00000238651', 'BIOTYPE': 'protein_coding', 'EXON': '3/3', 'INTRON': '', 'HGVSc': 'ENST00000238651.5:c.983A>G', 'HGVSp': 'ENSP00000238651.5:p.(His328Arg)', 'cDNA_position': '1165', 'CDS_position': '983', 'Protein_position': '328', 'Amino_acids': 'H/R', 'Codons': 'cAc/cGc', 'Existing_variation': 'rs149033118&COSV53149735', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '18431', 'CANONICAL': 'YES', 'RefSeq': 'NM_006821.5', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.3)', 'PolyPhen': 'benign(0.003)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.7745', 'gnomADe_AFR_AF': '0.7741', 'gnomADe_AMR_AF': '0.8902', 'gnomADe_ASJ_AF': '0.6742', 'gnomADe_EAS_AF': '0.9608', 'gnomADe_FIN_AF': '0.7895', 'gnomADe_NFE_AF': '0.7097', 'gnomADe_OTH_AF': '0.7437', 'gnomADe_SAS_AF': '0.7466', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': 'benign', 'SOMATIC': '0&1', 'PHENO': '1&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.026', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0686', 'am_protein_variant': 'H328R', 'am_transcript_id': 'ENST00000238651.5', 'am_uniprot_id': 'P49753', 'PrimateAI': '0.246645256877', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.212321123023013', 'pTriplo': '0.57137228846235', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'GOLGA5', 'Gene': 'ENSG00000066455', 'Feature_type': 'Transcript', 'Feature': 'ENST00000163416', 'BIOTYPE': 'protein_coding', 'EXON': '5/13', 'INTRON': '', 'HGVSc': 'ENST00000163416.2:c.1061T>C', 'HGVSp': 'ENSP00000163416.2:p.(Leu354Pro)', 'cDNA_position': '1317', 'CDS_position': '1061', 'Protein_position': '354', 'Amino_acids': 'L/P', 'Codons': 'cTg/cCg', 'Existing_variation': 'rs183339423', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '4428', 'CANONICAL': 'YES', 'RefSeq': 'NM_005113.3', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '1', 'SIFT': 'deleterious(0)', 'PolyPhen': 'probably_damaging(1)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0003903', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.004899', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0.0002674', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '0.738', 'am_class': 'pathogenic', 'am_genome': 'hg19', 'am_pathogenicity': '0.9405', 'am_protein_variant': 'L354P', 'am_transcript_id': 'ENST00000163416.2', 'am_uniprot_id': 'Q8TBA6', 'PrimateAI': '0.761112272739', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.742378171744482', 'pTriplo': '0.29567612089583', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'CEP170B', 'Gene': 'ENSG00000099814', 'Feature_type': 'Transcript', 'Feature': 'ENST00000414716', 'BIOTYPE': 'protein_coding', 'EXON': '12/19', 'INTRON': '', 'HGVSc': 'ENST00000414716.3:c.3080G>A', 'HGVSp': 'ENSP00000404151.2:p.(Arg1027Gln)', 'cDNA_position': '3308', 'CDS_position': '3080', 'Protein_position': '1027', 'Amino_acids': 'R/Q', 'Codons': 'cGg/cAg', 'Existing_variation': 'rs369195788', 'DISTANCE': '', 'STRAND': '1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20362', 'CANONICAL': 'YES', 'RefSeq': 'NM_001112726.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'deleterious(0.01)', 'PolyPhen': 'possibly_damaging(0.691)', 'HGVS_OFFSET': '', 'gnomADe_AF': '4.298e-05', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.0003435', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '2.951e-05', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '3.319e-05', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.177', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '0.482838332653', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.859080027131947', 'pTriplo': '0.998491552028188', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.11405A>G', 'HGVSp': 'ENSP00000353114.4:p.(Lys3802Arg)', 'cDNA_position': '11525', 'CDS_position': '11405', 'Protein_position': '3802', 'Amino_acids': 'K/R', 'Codons': 'aAa/aGa', 'Existing_variation': 'rs553656839&COSV60907531', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'deleterious(0.04)', 'PolyPhen': 'possibly_damaging(0.471)', 'HGVS_OFFSET': '', 'gnomADe_AF': '4.028e-06', 'gnomADe_AFR_AF': '0', 'gnomADe_AMR_AF': '0', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '8.87e-06', 'gnomADe_OTH_AF': '0', 'gnomADe_SAS_AF': '0', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.090', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1259', 'am_protein_variant': 'K3802R', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.240911692381', 'Enformer_SAD': '0.002567', 'Enformer_SAR': '0.001445', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.10469G>A', 'HGVSp': 'ENSP00000353114.4:p.(Arg3490Lys)', 'cDNA_position': '10589', 'CDS_position': '10469', 'Protein_position': '3490', 'Amino_acids': 'R/K', 'Codons': 'aGg/aAg', 'Existing_variation': 'rs199872023&COSV105200860', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0003628', 'gnomADe_AFR_AF': '0.001949', 'gnomADe_AMR_AF': '0.0004964', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.0004513', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '0.0001687', 'gnomADe_OTH_AF': '0.0003332', 'gnomADe_SAS_AF': '0.0004597', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.068', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0569', 'am_protein_variant': 'R3490K', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.246543973684', 'Enformer_SAD': '-0.001425', 'Enformer_SAR': '-0.000726', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.9650T>C', 'HGVSp': 'ENSP00000353114.4:p.(Leu3217Pro)', 'cDNA_position': '9770', 'CDS_position': '9650', 'Protein_position': '3217', 'Amino_acids': 'L/P', 'Codons': 'cTc/cCc', 'Existing_variation': 'rs200384326&COSV60906872', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.36)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.1619', 'gnomADe_AFR_AF': '0.0945', 'gnomADe_AMR_AF': '0.1028', 'gnomADe_ASJ_AF': '0.2224', 'gnomADe_EAS_AF': '0.04482', 'gnomADe_FIN_AF': '0.09626', 'gnomADe_NFE_AF': '0.2186', 'gnomADe_OTH_AF': '0.227', 'gnomADe_SAS_AF': '0.155', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': 'benign', 'SOMATIC': '0&1', 'PHENO': '1&1', 'PUBMED': '24033266', 'NMD': '', 'REVEL': '0.041', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.045', 'am_protein_variant': 'L3217P', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.1969152987', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.9625G>C', 'HGVSp': 'ENSP00000353114.4:p.(Val3209Leu)', 'cDNA_position': '9745', 'CDS_position': '9625', 'Protein_position': '3209', 'Amino_acids': 'V/L', 'Codons': 'Gtg/Ctg', 'Existing_variation': 'rs201181175&COSV104657525&COSV60910059', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.44)', 'PolyPhen': 'benign(0.007)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.05299', 'gnomADe_AFR_AF': '0.03695', 'gnomADe_AMR_AF': '0.04486', 'gnomADe_ASJ_AF': '0.07623', 'gnomADe_EAS_AF': '0.01665', 'gnomADe_FIN_AF': '0.01321', 'gnomADe_NFE_AF': '0.06964', 'gnomADe_OTH_AF': '0.1053', 'gnomADe_SAS_AF': '0.05288', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1&1', 'PHENO': '0&1&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.015', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1345', 'am_protein_variant': 'V3209L', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.211235642433', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.6538C>G', 'HGVSp': 'ENSP00000353114.4:p.(Pro2180Ala)', 'cDNA_position': '6658', 'CDS_position': '6538', 'Protein_position': '2180', 'Amino_acids': 'P/A', 'Codons': 'Cca/Gca', 'Existing_variation': 'rs201959909&COSV60910780', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.01709', 'gnomADe_AFR_AF': '0.1451', 'gnomADe_AMR_AF': '0.0126', 'gnomADe_ASJ_AF': '0.007827', 'gnomADe_EAS_AF': '0.04471', 'gnomADe_FIN_AF': '0.01197', 'gnomADe_NFE_AF': '0.002457', 'gnomADe_OTH_AF': '0.01046', 'gnomADe_SAS_AF': '0.00939', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.022', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0421', 'am_protein_variant': 'P2180A', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.188640147448', 'Enformer_SAD': '0.002258', 'Enformer_SAR': '0.002237', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.6357G>C', 'HGVSp': 'ENSP00000353114.4:p.(Met2119Ile)', 'cDNA_position': '6477', 'CDS_position': '6357', 'Protein_position': '2119', 'Amino_acids': 'M/I', 'Codons': 'atG/atC', 'Existing_variation': 'rs199873247&COSV60908251', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.25)', 'PolyPhen': 'benign(0.01)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.003935', 'gnomADe_AFR_AF': '0.01001', 'gnomADe_AMR_AF': '0.00163', 'gnomADe_ASJ_AF': '0.00549', 'gnomADe_EAS_AF': '0.006388', 'gnomADe_FIN_AF': '0.003112', 'gnomADe_NFE_AF': '0.001962', 'gnomADe_OTH_AF': '0.004411', 'gnomADe_SAS_AF': '0.008998', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.020', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.2526', 'am_protein_variant': 'M2119I', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.178342416883', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.6355A>G', 'HGVSp': 'ENSP00000353114.4:p.(Met2119Val)', 'cDNA_position': '6475', 'CDS_position': '6355', 'Protein_position': '2119', 'Amino_acids': 'M/V', 'Codons': 'Atg/Gtg', 'Existing_variation': 'rs199970974&COSV60908255', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0.001)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.008006', 'gnomADe_AFR_AF': '0.02801', 'gnomADe_AMR_AF': '0.003913', 'gnomADe_ASJ_AF': '0.01086', 'gnomADe_EAS_AF': '0.008006', 'gnomADe_FIN_AF': '0.00411', 'gnomADe_NFE_AF': '0.005113', 'gnomADe_OTH_AF': '0.008782', 'gnomADe_SAS_AF': '0.01394', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.008', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.0555', 'am_protein_variant': 'M2119V', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.186300992966', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'AHNAK2', 'Gene': 'ENSG00000185567', 'Feature_type': 'Transcript', 'Feature': 'ENST00000333244', 'BIOTYPE': 'protein_coding', 'EXON': '7/7', 'INTRON': '', 'HGVSc': 'ENST00000333244.5:c.3286C>A', 'HGVSp': 'ENSP00000353114.4:p.(Leu1096Ile)', 'cDNA_position': '3406', 'CDS_position': '3286', 'Protein_position': '1096', 'Amino_acids': 'L/I', 'Codons': 'Ctc/Atc', 'Existing_variation': 'rs201268922&COSV60908273', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': '', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '20125', 'CANONICAL': 'YES', 'RefSeq': 'NM_138420.2', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.74)', 'PolyPhen': 'benign(0.062)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.08405', 'gnomADe_AFR_AF': '0.2372', 'gnomADe_AMR_AF': '0.07078', 'gnomADe_ASJ_AF': '0.03393', 'gnomADe_EAS_AF': '0.2609', 'gnomADe_FIN_AF': '0.02585', 'gnomADe_NFE_AF': '0.03983', 'gnomADe_OTH_AF': '0.06948', 'gnomADe_SAS_AF': '0.1444', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '0&1', 'PHENO': '0&1', 'PUBMED': '', 'NMD': '', 'REVEL': '0.011', 'am_class': 'benign', 'am_genome': 'hg19', 'am_pathogenicity': '0.1063', 'am_protein_variant': 'L1096I', 'am_transcript_id': 'ENST00000333244.5', 'am_uniprot_id': 'Q8IVF2', 'PrimateAI': '0.185851708055', 'Enformer_SAD': '0.00351', 'Enformer_SAR': '0.004204', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '0.140993904006156', 'pTriplo': '0.0670228659834691', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHJ6', 'Gene': 'ENSG00000211900', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390560', 'BIOTYPE': 'IG_J_gene', 'EXON': '1/1', 'INTRON': '', 'HGVSc': 'ENST00000390560.2:c.17A>G', 'HGVSp': 'ENSP00000419223.1:p.(Tyr6Cys)', 'cDNA_position': '17', 'CDS_position': '17', 'Protein_position': '6', 'Amino_acids': 'Y/C', 'Codons': 'tAc/tGc', 'Existing_variation': 'rs2338627', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_start_NF&cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5540', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': '', 'PolyPhen': 'unknown(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '', 'gnomADe_AFR_AF': '', 'gnomADe_AMR_AF': '', 'gnomADe_ASJ_AF': '', 'gnomADe_EAS_AF': '', 'gnomADe_FIN_AF': '', 'gnomADe_NFE_AF': '', 'gnomADe_OTH_AF': '', 'gnomADe_SAS_AF': '', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'C', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV1-3', 'Gene': 'ENSG00000211935', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390595', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390595.2:c.187A>G', 'HGVSp': 'ENSP00000375004.2:p.(Arg63Gly)', 'cDNA_position': '229', 'CDS_position': '187', 'Protein_position': '63', 'Amino_acids': 'R/G', 'Codons': 'Agg/Ggg', 'Existing_variation': 'rs200773107', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5552', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0004538', 'gnomADe_AFR_AF': '0.0001426', 'gnomADe_AMR_AF': '0.0001792', 'gnomADe_ASJ_AF': '0.001569', 'gnomADe_EAS_AF': '0.0002668', 'gnomADe_FIN_AF': '0.0001191', 'gnomADe_NFE_AF': '0.0006554', 'gnomADe_OTH_AF': '0.0007874', 'gnomADe_SAS_AF': '0.0001284', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV1-3', 'Gene': 'ENSG00000211935', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390595', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390595.2:c.160C>A', 'HGVSp': 'ENSP00000375004.2:p.(His54Asn)', 'cDNA_position': '202', 'CDS_position': '160', 'Protein_position': '54', 'Amino_acids': 'H/N', 'Codons': 'Cat/Aat', 'Existing_variation': 'rs114502996', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5552', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.28)', 'PolyPhen': 'benign(0.012)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.1053', 'gnomADe_AFR_AF': '0.02729', 'gnomADe_AMR_AF': '0.1187', 'gnomADe_ASJ_AF': '0.1432', 'gnomADe_EAS_AF': '0.08918', 'gnomADe_FIN_AF': '0.06514', 'gnomADe_NFE_AF': '0.1058', 'gnomADe_OTH_AF': '0.08438', 'gnomADe_SAS_AF': '0.1733', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV1-3', 'Gene': 'ENSG00000211935', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390595', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390595.2:c.94A>C', 'HGVSp': 'ENSP00000375004.2:p.(Lys32Gln)', 'cDNA_position': '136', 'CDS_position': '94', 'Protein_position': '32', 'Amino_acids': 'K/Q', 'Codons': 'Aag/Cag', 'Existing_variation': 'rs186640498', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5552', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.08)', 'PolyPhen': 'benign(0.259)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.0008454', 'gnomADe_AFR_AF': '0.0001952', 'gnomADe_AMR_AF': '2.901e-05', 'gnomADe_ASJ_AF': '0', 'gnomADe_EAS_AF': '0.008137', 'gnomADe_FIN_AF': '0', 'gnomADe_NFE_AF': '5.448e-05', 'gnomADe_OTH_AF': '0.0009973', 'gnomADe_SAS_AF': '0.001504', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '0.000914', 'Enformer_SAR': '0.0006433', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV1-3', 'Gene': 'ENSG00000211935', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390595', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390595.2:c.88G>T', 'HGVSp': 'ENSP00000375004.2:p.(Val30Leu)', 'cDNA_position': '130', 'CDS_position': '88', 'Protein_position': '30', 'Amino_acids': 'V/L', 'Codons': 'Gtg/Ttg', 'Existing_variation': 'rs113475512', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5552', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.73)', 'PolyPhen': 'benign(0.073)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.03215', 'gnomADe_AFR_AF': '0.01088', 'gnomADe_AMR_AF': '0.02851', 'gnomADe_ASJ_AF': '0.02847', 'gnomADe_EAS_AF': '0.04032', 'gnomADe_FIN_AF': '0.02584', 'gnomADe_NFE_AF': '0.03722', 'gnomADe_OTH_AF': '0.01991', 'gnomADe_SAS_AF': '0.03571', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'A', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV1-3', 'Gene': 'ENSG00000211935', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390595', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390595.2:c.82G>T', 'HGVSp': 'ENSP00000375004.2:p.(Ala28Ser)', 'cDNA_position': '124', 'CDS_position': '82', 'Protein_position': '28', 'Amino_acids': 'A/S', 'Codons': 'Gct/Tct', 'Existing_variation': 'rs200143794', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5552', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated(0.07)', 'PolyPhen': 'benign(0.436)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.007379', 'gnomADe_AFR_AF': '0.002697', 'gnomADe_AMR_AF': '0.006147', 'gnomADe_ASJ_AF': '0.007439', 'gnomADe_EAS_AF': '0.0103', 'gnomADe_FIN_AF': '0.00605', 'gnomADe_NFE_AF': '0.009013', 'gnomADe_OTH_AF': '0.004168', 'gnomADe_SAS_AF': '0.00564', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV4-4', 'Gene': 'ENSG00000211936', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390596', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390596.2:c.103G>C', 'HGVSp': 'ENSP00000375005.2:p.(Glu35Gln)', 'cDNA_position': '167', 'CDS_position': '103', 'Protein_position': '35', 'Amino_acids': 'E/Q', 'Codons': 'Gag/Cag', 'Existing_variation': 'rs202003233', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5652', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated_low_confidence(1)', 'PolyPhen': 'benign(0.015)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.101', 'gnomADe_AFR_AF': '0.2128', 'gnomADe_AMR_AF': '0.08589', 'gnomADe_ASJ_AF': '0.1084', 'gnomADe_EAS_AF': '0.1155', 'gnomADe_FIN_AF': '0.07618', 'gnomADe_NFE_AF': '0.0876', 'gnomADe_OTH_AF': '0.08845', 'gnomADe_SAS_AF': '0.125', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '0.0002832', 'Enformer_SAR': '0.0002406', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV3-11', 'Gene': 'ENSG00000211941', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390601', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390601.2:c.232T>A', 'HGVSp': 'ENSP00000375010.2:p.(Tyr78Asn)', 'cDNA_position': '352', 'CDS_position': '232', 'Protein_position': '78', 'Amino_acids': 'Y/N', 'Codons': 'Tac/Aac', 'Existing_variation': 'rs201642285&rs782101832', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5580', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated_low_confidence(0.2)', 'PolyPhen': 'benign(0.019)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.2462', 'gnomADe_AFR_AF': '0.1041', 'gnomADe_AMR_AF': '0.2451', 'gnomADe_ASJ_AF': '0.248', 'gnomADe_EAS_AF': '0.1938', 'gnomADe_FIN_AF': '0.2416', 'gnomADe_NFE_AF': '0.2515', 'gnomADe_OTH_AF': '0.272', 'gnomADe_SAS_AF': '0.3553', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'G', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV3-11', 'Gene': 'ENSG00000211941', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390601', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390601.2:c.230T>C', 'HGVSp': 'ENSP00000375010.2:p.(Ile77Thr)', 'cDNA_position': '350', 'CDS_position': '230', 'Protein_position': '77', 'Amino_acids': 'I/T', 'Codons': 'aTa/aCa', 'Existing_variation': 'rs200148307&rs2047404730', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5580', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated_low_confidence(1)', 'PolyPhen': 'benign(0)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.2555', 'gnomADe_AFR_AF': '0.1065', 'gnomADe_AMR_AF': '0.255', 'gnomADe_ASJ_AF': '0.2585', 'gnomADe_EAS_AF': '0.199', 'gnomADe_FIN_AF': '0.2498', 'gnomADe_NFE_AF': '0.2621', 'gnomADe_OTH_AF': '0.2809', 'gnomADe_SAS_AF': '0.3671', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n",
      "{'Allele': 'T', 'Consequence': 'missense_variant', 'IMPACT': 'MODERATE', 'SYMBOL': 'IGHV3-11', 'Gene': 'ENSG00000211941', 'Feature_type': 'Transcript', 'Feature': 'ENST00000390601', 'BIOTYPE': 'IG_V_gene', 'EXON': '2/2', 'INTRON': '', 'HGVSc': 'ENST00000390601.2:c.227C>A', 'HGVSp': 'ENSP00000375010.2:p.(Thr76Asn)', 'cDNA_position': '347', 'CDS_position': '227', 'Protein_position': '76', 'Amino_acids': 'T/N', 'Codons': 'aCc/aAc', 'Existing_variation': 'rs1555434221&rs200917720&rs782053608', 'DISTANCE': '', 'STRAND': '-1', 'FLAGS': 'cds_end_NF', 'VARIANT_CLASS': 'SNV', 'SYMBOL_SOURCE': 'HGNC', 'HGNC_ID': '5580', 'CANONICAL': 'YES', 'RefSeq': '', 'REFSEQ_MATCH': '', 'SOURCE': 'Ensembl', 'REFSEQ_OFFSET': '', 'GENE_PHENO': '', 'SIFT': 'tolerated_low_confidence(0.24)', 'PolyPhen': 'benign(0.003)', 'HGVS_OFFSET': '', 'gnomADe_AF': '0.2611', 'gnomADe_AFR_AF': '0.08857', 'gnomADe_AMR_AF': '0.259', 'gnomADe_ASJ_AF': '0.2715', 'gnomADe_EAS_AF': '0.202', 'gnomADe_FIN_AF': '0.2579', 'gnomADe_NFE_AF': '0.2707', 'gnomADe_OTH_AF': '0.2881', 'gnomADe_SAS_AF': '0.3749', 'gnomADg_AF': '', 'gnomADg_AFR_AF': '', 'gnomADg_AMI_AF': '', 'gnomADg_AMR_AF': '', 'gnomADg_ASJ_AF': '', 'gnomADg_EAS_AF': '', 'gnomADg_FIN_AF': '', 'gnomADg_MID_AF': '', 'gnomADg_NFE_AF': '', 'gnomADg_OTH_AF': '', 'gnomADg_SAS_AF': '', 'CLIN_SIG': '', 'SOMATIC': '', 'PHENO': '', 'PUBMED': '', 'NMD': '', 'REVEL': '', 'am_class': '', 'am_genome': '', 'am_pathogenicity': '', 'am_protein_variant': '', 'am_transcript_id': '', 'am_uniprot_id': '', 'PrimateAI': '', 'Enformer_SAD': '', 'Enformer_SAR': '', '5UTR_annotation': '', '5UTR_consequence': '', 'Existing_InFrame_oORFs': '', 'Existing_OutOfFrame_oORFs': '', 'Existing_uORFs': '', 'pHaplo': '', 'pTriplo': '', 'TSSDistance\"': ''}\n"
     ]
    }
   ],
   "source": [
    "stop = 50\n",
    "vcf = cyvcf2.VCF(raw_vcf)\n",
    "i = 0\n",
    "for variant in vcf:\n",
    "    csq_values: list = variant.INFO.get('CSQ').split('|')\n",
    "    csq_kv: dict = dict(zip(csq_keys, csq_values))\n",
    "    if csq_kv['Consequence'] == 'missense_variant':\n",
    "        print(csq_kv)\n",
    "        i += 1\n",
    "    \n",
    "    if i == stop:\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Allele': 'A',\n",
       " 'Consequence': 'intergenic_variant',\n",
       " 'IMPACT': 'MODIFIER',\n",
       " 'SYMBOL': '',\n",
       " 'Gene': '',\n",
       " 'Feature_type': '',\n",
       " 'Feature': '',\n",
       " 'BIOTYPE': '',\n",
       " 'EXON': '',\n",
       " 'INTRON': '',\n",
       " 'HGVSc': '',\n",
       " 'HGVSp': '',\n",
       " 'cDNA_position': '',\n",
       " 'CDS_position': '',\n",
       " 'Protein_position': '',\n",
       " 'Amino_acids': '',\n",
       " 'Codons': '',\n",
       " 'Existing_variation': 'rs1829082',\n",
       " 'DISTANCE': '',\n",
       " 'STRAND': '',\n",
       " 'FLAGS': '',\n",
       " 'VARIANT_CLASS': 'SNV',\n",
       " 'SYMBOL_SOURCE': '',\n",
       " 'HGNC_ID': '',\n",
       " 'CANONICAL': '',\n",
       " 'RefSeq': '',\n",
       " 'REFSEQ_MATCH': '',\n",
       " 'SOURCE': '',\n",
       " 'REFSEQ_OFFSET': '',\n",
       " 'GENE_PHENO': '',\n",
       " 'SIFT': '',\n",
       " 'PolyPhen': '',\n",
       " 'HGVS_OFFSET': '',\n",
       " 'gnomADe_AF': '',\n",
       " 'gnomADe_AFR_AF': '',\n",
       " 'gnomADe_AMR_AF': '',\n",
       " 'gnomADe_ASJ_AF': '',\n",
       " 'gnomADe_EAS_AF': '',\n",
       " 'gnomADe_FIN_AF': '',\n",
       " 'gnomADe_NFE_AF': '',\n",
       " 'gnomADe_OTH_AF': '',\n",
       " 'gnomADe_SAS_AF': '',\n",
       " 'gnomADg_AF': '',\n",
       " 'gnomADg_AFR_AF': '',\n",
       " 'gnomADg_AMI_AF': '',\n",
       " 'gnomADg_AMR_AF': '',\n",
       " 'gnomADg_ASJ_AF': '',\n",
       " 'gnomADg_EAS_AF': '',\n",
       " 'gnomADg_FIN_AF': '',\n",
       " 'gnomADg_MID_AF': '',\n",
       " 'gnomADg_NFE_AF': '',\n",
       " 'gnomADg_OTH_AF': '',\n",
       " 'gnomADg_SAS_AF': '',\n",
       " 'CLIN_SIG': '',\n",
       " 'SOMATIC': '',\n",
       " 'PHENO': '',\n",
       " 'PUBMED': '',\n",
       " 'NMD': '',\n",
       " 'REVEL': '',\n",
       " 'am_class': '',\n",
       " 'am_genome': '',\n",
       " 'am_pathogenicity': '',\n",
       " 'am_protein_variant': '',\n",
       " 'am_transcript_id': '',\n",
       " 'am_uniprot_id': '',\n",
       " 'PrimateAI': '',\n",
       " 'Enformer_SAD': '',\n",
       " 'Enformer_SAR': '',\n",
       " '5UTR_annotation': '',\n",
       " '5UTR_consequence': '',\n",
       " 'Existing_InFrame_oORFs': '',\n",
       " 'Existing_OutOfFrame_oORFs': '',\n",
       " 'Existing_uORFs': '',\n",
       " 'pHaplo': '',\n",
       " 'pTriplo': '',\n",
       " 'TSSDistance\"': ''}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csq_kv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf = cyvcf2.VCF(raw_vcf)\n",
    "i = 0\n",
    "while i < 20:\n",
    "    i += 1\n",
    "    v = next(vcf)\n",
    "\n",
    "    # VCF like information\n",
    "    chrom: str = v.CHROM\n",
    "    pos: int = v.POS\n",
    "    ref_allele: str = v.REF\n",
    "    gt_allele: str = v.ALT\n",
    "    \n",
    "    \n",
    "    # Parsing INFO\n",
    "    gt_types_raw = v.gt_types\n",
    "    if gt_types_raw == 0:\n",
    "        gt_types = \"Homozygous Reference\"\n",
    "    elif gt_types_raw == 1:\n",
    "        gt_types = \"Heterozygous\"\n",
    "    elif gt_types_raw == 2:\n",
    "        gt_types = \"Unknown\"\n",
    "    elif gt_types_raw == 3:\n",
    "        gt_types = \"Homozygous Alternate\"\n",
    "    else: \n",
    "        gt_types = \"No Call\"\n",
    "\n",
    "    gt_re_depths = v.gt_ref_depths\n",
    "    gt_alt_depths = v.gt_alt_depths\n",
    "    gt_phases = v.gt_phases\n",
    "    gt_quals = v.gt_quals\n",
    "    gt_bases = v.gt_bases\n",
    "\n",
    "    gt_allele_balances = np.divide(gt_alt_depths, gt_re_depths + gt_alt_depths)\n",
    "    print(f\"AB: {gt_allele_balances} = {gt_alt_depths} / {gt_re_depths + gt_alt_depths}\")    \n",
    "    gt = v.genotypes\n",
    "\n",
    "    print(f\"GT_types: {v.gt_types}, GT_ref_depths: {v.gt_ref_depths}, GT_alt_depths: {v.gt_alt_depths}, GT_phases: {v.gt_phases}, GT_quals: {v.gt_quals}, GT_bases: {v.gt_bases}\")\n",
    "    # \"AD\", \"DP\", \"GQ\", \"GT\", \"PL\", \"JpnAD2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'numpy.int32'>\n"
     ]
    }
   ],
   "source": [
    "variant = next(vcf)\n",
    "variant.gt_types\n",
    "\n",
    "print(type(variant.gt_types[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'HeaderType': 'GENERIC'}\n",
      "{'ID': 'PASS', 'Description': '\"All filters passed\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': '1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '2', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '3', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '4', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '5', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '6', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '7', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '8', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '9', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '10', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '11', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '12', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '13', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '14', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '15', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '16', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '17', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '18', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '19', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '20', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '21', 'HeaderType': 'CONTIG'}\n",
      "{'ID': '22', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'X', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'Y', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'MT', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000207.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000226.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000229.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000231.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000210.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000239.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000235.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000201.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000247.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000245.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000197.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000203.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000246.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000249.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000196.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000248.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000244.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000238.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000202.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000234.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000232.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000206.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000240.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000236.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000241.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000243.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000242.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000230.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000237.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000233.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000204.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000198.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000208.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000191.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000227.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000228.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000214.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000221.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000209.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000218.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000220.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000213.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000211.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000199.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000217.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000216.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000215.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000205.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000219.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000224.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000223.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000195.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000212.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000222.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000200.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000193.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000194.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000225.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'GL000192.1', 'HeaderType': 'CONTIG'}\n",
      "{'ID': 'FS200', 'Description': '\"FS > 200.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'FS60', 'Description': '\"FS > 60.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'LowQual', 'Description': '\"Low quality\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'MQ40', 'Description': '\"MQ < 40.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'MQRankSum-12.5', 'Description': '\"MQRankSum < -12.5\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'QD2.0', 'Description': '\"QD < 2.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'ReadPosRankSum-20.0', 'Description': '\"ReadPosRankSum < -20.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'ReadPosRankSum-8.0', 'Description': '\"ReadPosRankSum < -8.0\"', 'HeaderType': 'FILTER'}\n",
      "{'ID': 'dbSNP', 'Description': '\"DB\"', 'HeaderType': 'FILTER'}\n",
      "{'Type': 'Integer', 'Number': 'A', 'ID': 'AC', 'Description': '\"Allele count in genotypes, for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': 'A', 'ID': 'AF', 'Description': '\"Allele Frequency, for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'AN', 'Description': '\"Total number of alleles in called genotypes\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'BaseQRankSum', 'Description': '\"Z-score from Wilcoxon rank sum test of Alt Vs. Ref base qualities\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Flag', 'Number': '0', 'ID': 'DB', 'Description': '\"dbSNP Membership\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'DP', 'Description': '\"Approximate read depth; some reads may have been filtered\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Flag', 'Number': '0', 'ID': 'DS', 'Description': '\"Were any of the samples downsampled?\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'Dels', 'Description': '\"Fraction of Reads Containing Spanning Deletions\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ExcessHet', 'Description': '\"Phred-scaled p-value for exact test of excess heterozygosity\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'FS', 'Description': '\"Phred-scaled p-value using Fisher\\'s exact test to detect strand bias\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'HaplotypeScore', 'Description': '\"Consistency of the site with at most two segregating haplotypes\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'InbreedingCoeff', 'Description': '\"Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hardy-Weinberg expectation\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'String', 'Number': '3', 'ID': 'JpnMutation', 'Description': '\"Normal Control Allele Prevalence\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': 'A', 'ID': 'MLEAC', 'Description': '\"Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': 'A', 'ID': 'MLEAF', 'Description': '\"Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'MQ', 'Description': '\"RMS Mapping Quality\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'MQ0', 'Description': '\"Total Mapping Quality Zero Reads\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'MQRankSum', 'Description': '\"Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'QD', 'Description': '\"Variant Confidence/Quality by Depth\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '.', 'ID': 'RPA', 'Description': '\"Number of times tandem repeat unit is repeated, for each allele (including reference)\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'String', 'Number': '1', 'ID': 'RU', 'Description': '\"Tandem repeat unit (bases)\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ReadPosRankSum', 'Description': '\"Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'SOR', 'Description': '\"Symmetric Odds Ratio of 2x2 contingency table to detect strand bias\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Flag', 'Number': '0', 'ID': 'STR', 'Description': '\"Variant is a short tandem repeat\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Flag', 'Number': '0', 'ID': 'dbSNPcommon', 'Description': '\"calculated by flag of overlapping values in field COMMON from /betelgeuse04/analysis/utsu/resources/vcfanno/dbSNP-b156/GCF_000001405.25.singleALT.common.chrmod.sort.vcf.gz\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'String', 'Number': '1', 'ID': 'set', 'Description': '\"Source VCF for the merged record in CombineVariants\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': 'R', 'ID': 'AD', 'Description': '\"Allelic depths for the ref and alt alleles in the order listed\"', 'HeaderType': 'FORMAT'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'DP', 'Description': '\"Approximate read depth (reads with MQ=255 or with bad mates are filtered)\"', 'HeaderType': 'FORMAT'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'GQ', 'Description': '\"Genotype Quality\"', 'HeaderType': 'FORMAT'}\n",
      "{'Type': 'String', 'Number': '1', 'ID': 'GT', 'Description': '\"Genotype\"', 'HeaderType': 'FORMAT'}\n",
      "{'Type': 'String', 'Number': '1', 'ID': 'JpnAD2', 'Description': '\"Japanese mutation allele depth\"', 'HeaderType': 'FORMAT'}\n",
      "{'Type': 'Integer', 'Number': 'G', 'ID': 'PL', 'Description': '\"Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification\"', 'HeaderType': 'FORMAT'}\n",
      "{'ID': 'CombineVariants', 'HeaderType': 'STR'}\n",
      "{'ID': 'SelectVariants', 'HeaderType': 'STR'}\n",
      "{'ID': 'SelectVariants', 'HeaderType': 'STR'}\n",
      "{'ID': 'UnifiedGenotyper', 'HeaderType': 'STR'}\n",
      "{'ID': 'VariantFiltration', 'HeaderType': 'STR'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'gnomad4_remap37_af_grpmax_joint', 'Description': '\"added by echtvar Maximum allele frequency across genetic ancestry groups in joint dataset\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'gnomad4_remap37_nhomalt_joint', 'Description': '\"added by echtvar Count of homozygous individuals in the genetic ancestry group with the maximum allele frequency in joint dataset\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'tommo54kjpn_lifted37_af', 'Description': '\"added by echtvar Allele frequency, for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'tommo54kjpn_lifted37_nhomalt', 'Description': '\"added by echtvar Count of homozygous individuals\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'rgcme_remap37_af_all', 'Description': '\"added by echtvar Alternate allele frequency, ALL\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ncbn_remap37_af_jpn', 'Description': '\"added by echtvar Allele frequency in JPN\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ncbn_remap37_af_jpn_hondo', 'Description': '\"added by echtvar Allele frequency in JPN-Hondo\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ncbn_remap37_af_jpn_ryukyu', 'Description': '\"added by echtvar Allele frequency in JPN-Ryukyu\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhom_jpn', 'Description': '\"added by echtvar Allele counts in homozygous genotypes in JPN\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhom_jpn_hondo', 'Description': '\"added by echtvar Allele counts in homozygous genotypes in JPN-Hondo\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhom_jpn_ryukyu', 'Description': '\"added by echtvar Allele counts in homozygous genotypes in JPN-Ryukyu\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhemi_jpn', 'Description': '\"added by echtvar Allele counts in hemizygous genotypes in JPN\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhemi_jpn_hondo', 'Description': '\"added by echtvar Allele counts in hemizygous genotypes in JPN-Hondo\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ncbn_remap37_nhemi_jpn_ryukyu', 'Description': '\"added by echtvar Allele counts in hemizygous genotypes in JPN-Ryukyu\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af', 'Description': '\"added by echtvar Allele Frequency, for each ALT allele, in the same order as listed\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_oce', 'Description': '\"added by echtvar Oceania Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_oce', 'Description': '\"added by echtvar Oceania Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_nea', 'Description': '\"added by echtvar North East Asian Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_nea', 'Description': '\"added by echtvar North East Asian Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_sea', 'Description': '\"added by echtvar South East Asian Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_sea', 'Description': '\"added by echtvar South East Asian Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_sas', 'Description': '\"added by echtvar South Asian Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_sas', 'Description': '\"added by echtvar South Asian Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_afr', 'Description': '\"added by echtvar African Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_afr', 'Description': '\"added by echtvar African Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_amr', 'Description': '\"added by echtvar American Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_amr', 'Description': '\"added by echtvar American Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Float', 'Number': '1', 'ID': 'ga_af_wer', 'Description': '\"added by echtvar West Eurasia Alt Allele Frequency\"', 'HeaderType': 'INFO'}\n",
      "{'Type': 'Integer', 'Number': '1', 'ID': 'ga_nhom_wer', 'Description': '\"added by echtvar West Eurasia Homozygous Sample Counts\"', 'HeaderType': 'INFO'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'Type': 'String', 'Number': '.', 'ID': 'SpliceAI', 'Description': '\"SpliceAIv1.3.1 variant annotation. These include delta scores (DS) and delta positions (DP) for acceptor gain (AG), acceptor loss (AL), donor gain (DG), and donor loss (DL). Format: ALLELE|SYMBOL|DS_AG|DS_AL|DS_DG|DS_DL|DP_AG|DP_AL|DP_DG|DP_DL\"', 'HeaderType': 'INFO'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'Type': 'String', 'Number': '.', 'ID': 'CSQ', 'Description': '\"Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|VARIANT_CLASS|SYMBOL_SOURCE|HGNC_ID|CANONICAL|RefSeq|REFSEQ_MATCH|SOURCE|REFSEQ_OFFSET|GENE_PHENO|SIFT|PolyPhen|HGVS_OFFSET|gnomADe_AF|gnomADe_AFR_AF|gnomADe_AMR_AF|gnomADe_ASJ_AF|gnomADe_EAS_AF|gnomADe_FIN_AF|gnomADe_NFE_AF|gnomADe_OTH_AF|gnomADe_SAS_AF|gnomADg_AF|gnomADg_AFR_AF|gnomADg_AMI_AF|gnomADg_AMR_AF|gnomADg_ASJ_AF|gnomADg_EAS_AF|gnomADg_FIN_AF|gnomADg_MID_AF|gnomADg_NFE_AF|gnomADg_OTH_AF|gnomADg_SAS_AF|CLIN_SIG|SOMATIC|PHENO|PUBMED|NMD|REVEL|am_class|am_genome|am_pathogenicity|am_protein_variant|am_transcript_id|am_uniprot_id|PrimateAI|Enformer_SAD|Enformer_SAR|5UTR_annotation|5UTR_consequence|Existing_InFrame_oORFs|Existing_OutOfFrame_oORFs|Existing_uORFs|pHaplo|pTriplo|TSSDistance\"', 'HeaderType': 'INFO'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n",
      "{'HeaderType': 'GENERIC'}\n"
     ]
    }
   ],
   "source": [
    "for h in vcf.header_iter():\n",
    "    print(h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "csq_keys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<bound method Variant.format of Variant(Y:3447245 G/T)>\n",
      "<bound method Variant.format of Variant(Y:4968655 T/G)>\n",
      "<bound method Variant.format of Variant(Y:5441327 G/A)>\n",
      "<bound method Variant.format of Variant(Y:5441332 C/CT)>\n",
      "<bound method Variant.format of Variant(Y:7209078 C/G)>\n",
      "<bound method Variant.format of Variant(Y:7556207 G/A)>\n",
      "<bound method Variant.format of Variant(Y:8805525 A/G)>\n",
      "<bound method Variant.format of Variant(Y:9004556 C/T)>\n",
      "<bound method Variant.format of Variant(Y:9385720 CG/C)>\n",
      "<bound method Variant.format of Variant(Y:9920543 G/A)>\n",
      "<bound method Variant.format of Variant(Y:9930004 C/A)>\n",
      "<bound method Variant.format of Variant(Y:9930025 A/G)>\n",
      "<bound method Variant.format of Variant(Y:9930055 A/G)>\n",
      "<bound method Variant.format of Variant(Y:9930057 G/A)>\n",
      "<bound method Variant.format of Variant(Y:9930095 C/T)>\n",
      "<bound method Variant.format of Variant(Y:9930110 G/C)>\n",
      "<bound method Variant.format of Variant(Y:9930129 C/T)>\n",
      "<bound method Variant.format of Variant(Y:9930160 A/G)>\n",
      "<bound method Variant.format of Variant(Y:9930174 A/G)>\n",
      "<bound method Variant.format of Variant(Y:9930192 T/G)>\n"
     ]
    }
   ],
   "source": [
    "vcf = cyvcf2.VCF(raw_vcf)\n",
    "i = 0\n",
    "while i < 20:\n",
    "    i += 1\n",
    "    variant = next(vcf)\n",
    "    print(variant.format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the current cell or a previous cell. \n",
      "\u001b[1;31mPlease review the code in the cell(s) to identify a possible cause of the failure. \n",
      "\u001b[1;31mClick <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. \n",
      "\u001b[1;31mView Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "h"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "vep",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
